web_poet/rules.py

```diff
@@ -0,0 +1,338 @@
+
+        This is useful in cases wherein you need to store some selected rules
+        from multiple external packages. See this :ref:`example
+        <overrides-custom-registry>`.
+        """
+        return cls({rule.use: rule for rule in rules})
+
+    @classmethod
+    def from_override_rules(
+        cls: Type[PageObjectRegistryTV], rules: List[ApplyRule]
+    ) -> PageObjectRegistryTV:
+        """Deprecated. Use :meth:`~.PageObjectRegistry.from_apply_rules` instead."""
+        msg = (
+            "The 'from_override_rules' method is deprecated. "
+            "Use 'from_apply_rules' instead."
+        )
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
+        return cls.from_apply_rules(rules)
+
+    def handle_urls(
+        self,
+        include: Strings,
+        *,
+        overrides: Optional[Type[ItemPage]] = None,
+        instead_of: Optional[Type[ItemPage]] = None,
+        to_return: Optional[Type] = None,
+        exclude: Optional[Strings] = None,
+        priority: int = 500,
+        **kwargs,
+    ):
+        """
+        Class decorator that indicates that the decorated Page Object should be
+        used instead of the overridden one for a particular set the URLs.
+
+        The Page Object that is **overridden** is declared using the ``instead_of``
+        parameter.
+
+        The Item Class could also be **overridden** using the ``to_return``
+        parameter.
+
+        The **override** mechanism only works on certain URLs that match the
+        ``include`` and ``exclude`` parameters. See the documentation of the
+        `url-matcher <https://url-matcher.readthedocs.io/>`_ package for more
+        information about them.
+
+        Any extra parameters are stored as meta information that can be later used.
+
+        :param include: The URLs that should be handled by the decorated Page Object.
+        :param instead_of: The Page Object that should be `replaced`.
+        :param to_return: The Item Class holding the data returned by the Page Object.
+            This could be omitted as it could be derived from the ``Returns[ItemClass]``
+            or ``ItemPage[ItemClass]`` declaration of the Page Object. See
+            :ref:`item-classes` section. Code example in :ref:`combination` subsection.
+        :param exclude: The URLs over which the override should **not** happen.
+        :param priority: The resolution priority in case of `conflicting` rules.
+            A conflict happens when the ``include``, ``override``, and ``exclude``
+            parameters are the same. If so, the `highest priority` will be
+            chosen.
+        """
+
+        def wrapper(cls):
+
+            if overrides is not None:
+                msg = (
+                    "The 'overrides' parameter in @handle_urls is deprecated. "
+                    "Use the 'instead_of' parameter."
+                )
+                warnings.warn(msg, DeprecationWarning, stacklevel=2)
```

Comment:
Could you please check that this stacklevel points to the user code? I can't wrap my head around it, but suspect stacklevel=3 could be needed here :)

Reply:
If we have the following code:

```python
from web_poet import handle_urls

class Foo:
    ...

def do_something():
    @handle_urls("example.com", overrides=Foo)
    class Bar:
        ...

do_something()
```

**stacklevel=2** would have:
```python
tmp.py:10: DeprecationWarning: The 'overrides' parameter in @handle_urls is deprecated. Use the 'instead_of' parameter.
  class Bar:
```

**stacklevel=3** would have:
```python
tmp.py:14: DeprecationWarning: The 'overrides' parameter in @handle_urls is deprecated. Use the 'instead_of' parameter.
  do_something()
```

I think **stacklevel=2** is more appropriate here as it points to the class being decorated.

Reply:
Thanks for checking it, makes sense!